### PR TITLE
New: On Episode File Delete For Upgrade notification option

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
YES 153

#### Description
To add an option for notifications on episode file delete for upgrades

#### Todos
- [x] Tests
- [x] Wiki Updates

### CODE REVIEW

File: frontend/src/Settings/Notifications/Notifications/EditNotificationModalContent.js [hunk 1]
```diff
@@ -180,6 +182,22 @@ function EditNotificationModalContent(props) {
                 />
               </FormGroup>
 
+              {
+                onEpisodeFileDelete.value &&
```
Comment (markus101) [inline frontend/src/Settings/Notifications/Notifications/EditNotificationModalContent.js:186]: This should be a ternary

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 2]
```diff
@@ -136,6 +138,14 @@ class Notification extends Component {
             null
         }
 
+        {
+          supportsOnEpisodeFileDeleteForUpgrade && onEpisodeFileDelete && onEpisodeFileDeleteForUpgrade ?
+            <Label kind={kinds.SUCCESS}>
+              One Episode File Delete For Upgrade
```
Comment (markus101) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:144]: On

File: src/NzbDrone.Core/Datastore/Migration/153_add_on_episodefiledelete_for_upgrade.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(153)]
+    public class add_on_episodefiledelete_for_upgrade : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Alter.Table("Notifications").AddColumn("OnEpisodeFileDeleteForUpgrade").AsBoolean().WithDefaultValue(0);
```
Comment (markus101) [inline src/NzbDrone.Core/Datastore/Migration/153_add_on_episodefiledelete_for_upgrade.cs:11]: Default value should be 1 to maintain existing behaviour.

**Review by markus101 — CHANGES_REQUESTED** [review]
Just a few minor things

File: frontend/src/Settings/Notifications/Notifications/Notification.js (hunk 2 above)
Comment (RobinDadswell) [reply frontend/src/Settings/Notifications/Notifications/Notification.js:144]: Good catch - updated

File: src/NzbDrone.Core/Datastore/Migration/153_add_on_episodefiledelete_for_upgrade.cs (hunk 3 above)
Comment (RobinDadswell) [reply src/NzbDrone.Core/Datastore/Migration/153_add_on_episodefiledelete_for_upgrade.cs:11]: updated

**Comment by RobinDadswell** [comment]
@markus101  all those requested changes have been made and are in

**Pull request merged**